src/Couchbase.Lite.Shared/API/Database/Collection.cs

```diff
@@ -87,78 +147,226 @@ public void RemoveChangeListener(ListenerToken token)
+                }
+
+                _c4coll = (C4Collection*)LiteCoreBridge.Check(err =>
+                {
+                    return Native.c4db_createCollection(c4Db, collectionSpec, err);
+                });
+
+                return _c4coll != null;
+            }
+        }
+
+        internal bool DeleteCollection()
+        {
+            bool deleteSuccessful;
+            using (var collName_ = new C4String(Name))
+            using (var scopeName_ = new C4String(Scope?.Name)) {
+                var collectionSpec = new C4CollectionSpec() {
+                    name = collName_.AsFLSlice(),
+                    scope = scopeName_.AsFLSlice()
+                };
+
+                deleteSuccessful = (bool)LiteCoreBridge.Check(err =>
+                {
+                    return Native.c4db_deleteCollection(c4Db, collectionSpec, err);
+                });
+            }
+
+            return deleteSuccessful;
+        }
+
+        internal bool HasCollection()
+        {
+            bool hasCollection;
+            using (var collName_ = new C4String(Name))
+            using (var scopeName_ = new C4String(Scope?.Name)) {
+                var collectionSpec = new C4CollectionSpec() {
+                    name = collName_.AsFLSlice(),
+                    scope = scopeName_.AsFLSlice()
+                };
+
+                hasCollection = ThreadSafety.DoLocked(() =>
+                {
+                    return Native.c4db_hasCollection(c4Db, collectionSpec);
+                });
+            }
+
+            return hasCollection;
+        }
+
+        #endregion
+
+        #region Private Methods
+
+        internal Scope GetScope(string scope)
+        {
+            var s = Database.GetScope(scope);
+            if (s == null)
+                s = new Scope(Database, scope);
+
+            return s;
+        }
+
+        #endregion
+
+        #region object
+        /// <inheritdoc />
+        public override int GetHashCode()
+        {
+            return Name?.GetHashCode() ?? 0;
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            if (!(obj is Collection other)) {
+                return false;
+            }
+
+            return String.Equals(Name, other.Name, StringComparison.Ordinal)
+                && String.Equals(Scope.Name, other.Scope.Name, StringComparison.Ordinal);
+        }
+
+        /// <inheritdoc />
+        public override string ToString() => $"COLLECTION[{Name}] of SCOPE[{Scope.Name}]";
+        #endregion
+
+        #region IDisposable
 
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
         public void Dispose()
         {
-            throw new NotImplementedException();
+            ThreadSafety.DoLocked(() =>
+            {
+                _disposalWatchdog.Dispose();
+                Native.c4coll_release(_c4coll);
```

Comment:
As it stands this is an unbalanced release.  The value from getCollection is *unretained*